internal/spokes/spokes.go

```diff
@@ -367,7 +527,7 @@ func (r *spokesReceivePack) performReferenceDiscovery(ctx context.Context) error
 	}
 
 	// Collect the reference tips present in the parent repo in case this is a fork
-	parentRepoId := strings.TrimPrefix(os.Getenv("GIT_SOCKSTAT_VAR_parent_repo_id"), "uint:")
+	parentRepoId := os.Getenv("GIT_SOCKSTAT_VAR_parent_repo_id")
```

Comment:
Note: this is intentionally staying as a reference to an env var and reverting #65, rather than going through the `sockstat` package. This reintroduces a bug that is currently active in prod and that effectively disables `GIT_NW_ADVERTISE_REFS`. However, we want the bug fix (the rest of the changes in this file) to be behind a feature flag because the fix could end up causing problems on large repositories if we fix it incorrectly.